src/marten/conf/global_settings/database.cr

```diff
@@ -92,6 +92,35 @@ module Marten
         def initialize(@id : String)
         end
 
+        def from_url(url : String)
+          # URI.parse cant parse 'sqlite://:memory:'
+          if url.starts_with? "sqlite://:memory:"
+            self.backend = "sqlite"
+            self.name = ":memory:"
+            return
+          end
+
+          uri = URI.parse url
+
+          self.backend = uri.scheme
+          self.host = uri.host
+          self.port = uri.port
+          self.user = uri.user
+          self.password = uri.password
+          self.name = uri.scheme == "sqlite" ? uri.host : uri.path[1..]?
+
+          params_map = uri.query_params.to_h
+          
+          self.checkout_timeout = params_map.delete("checkout_timeout").try &.to_f64 || @checkout_timeout
```

Comment:
Would you want to raise a specific exception here in case one of these casts fails?